CADETProcess/optimization/surrogate.py

```diff
@@ -0,0 +1,381 @@
+from pathlib import Path
+from copy import deepcopy
+
+import numpy as np
+from matplotlib import pyplot as plt
+from sklearn.gaussian_process import (
+    GaussianProcessRegressor, GaussianProcessClassifier)
+from sklearn.base import BaseEstimator
+import hopsy
+
+from CADETProcess.optimization import (
+    OptimizationProblem, TrustConstr, SLSQP, OptimizationResults
+)
+
+class Surrogate:
+    """
+    Surrogate class for optimization problems.
+    """
+
+    def __init__(
+        self,
+        optimization_problem: OptimizationProblem,
```

Comment:
Only use OptimizationResults as parameter. The OptimizationProblem is also stored in that. This way, we can avoid inconsistencies.